test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -6262,6 +6262,52 @@ def forward(self, y):
                       input_names=["input_1"],
                       dynamic_axes={"input_1": [0, 1]})
 
+    @skipIfUnsupportedMinOpsetVersion(14)  # Need onnx::identity of sequence in opset 14
+    @skipIfONNXShapeInference(False)
+    def test_uninitialized_tensorList(self):
+        class UninitializedTensorListModel(torch.nn.Module):
+            def forward(self, y):
+                if y[0].shape[0] < 5:
+                    if y.size(0) == 1:
+                        y = y + 4
+                    else:
+                        return [y]
+                return [y]
+
+        y = torch.ones((3, 4), dtype=torch.int)
+        self.run_test(torch.jit.script(UninitializedTensorListModel()), y)
+
+    @skipIfUnsupportedMinOpsetVersion(14)  # Need onnx::identity of sequence in opset 14
+    @skipIfONNXShapeInference(False)
+    def test_uninitialized_tensorList_dynamic(self):
+        class UninitializedModel(torch.nn.Module):
+            def forward(self, y):
+                if y.shape[1] < 5:
+                    if y.size(0) == 1:
+                        y = y + 4
+                    else:
+                        return [y]
+                return [y]
```

Comment:
Can we have a unit test, if branch returns tensor list of empty list, else branch returns tensor list of several elements?